Fix: text item is created when creating a new empty package

plamut opened the pull request:
Resolves [SD-2870](https://dev.sourcefabric.org/browse/SD-2870).

amagdas commented:
Tests :blush:

plamut commented:
LOL.

Will add an e2e test.

Aca-jov commented:
Looks fine, except for missing tests :)

pavlovicnemanja92 commented:
Sorry, accidentally closed... Just add tests :+1:

sivakuna-aap commented:
:+1:

plamut commented:
I updated the commit with a test. I also added some new instructions to README, because it was not obvious how to run the tests in the first place.

amagdas commented:
:+1:

plamut commented:
@pavlovicnemanja92 We'll let you merge this PR to fulfil your long-lasting dream. :wink:

pavlovicnemanja92 commented:
Oh @plamut , my dream come true the moment I write to you "please add some tests" :)

ioanpocol commented on client/spec/authoring_spec.js:
```diff
@@ -102,4 +102,29 @@ describe('authoring', function() {
+            //
+            authoring.openAuthoringHome();
+
+            $('.navbtn .sd-create-btn').click();
```
Better to user PageObjects style to write the tests, see:
http://www.protractortest.org/#/page-objects

plamut replied:
Makes a lot of sense, I'll refactor. Thanks for the suggestion.

plamut commented:
Test refactored to use PageObjects.

ioanpocol commented on client/spec/authoring_spec.js:
```diff
@@ -102,4 +102,30 @@ describe('authoring', function() {
+
+            authoring.newEmptyPackageLink.click().then(function () {
+                return authoring.infoIconsBox.all(
+                    by.css('.filetype-icon-composite')
```
This selector by css is still on spec file

plamut replied:
Yeah, that's on purpose. The package icon is not some common, always present page element, it's specific to this test case. It's a dynamic thing that might or might not exist as a result of previous actions, and since it also represents the meat of the test, it is referenced explicitly.

FWIW, I saw the same pattern in the existing tests...

ioanpocol replied:
The basic idea is to move all selections on PageObjects. You can create a method that returns the list of elements filtered by the following css filter:  ".filetype-icon-composite".
And yes, some tests don't respect PageObjects pattern. We need to update them, but for sure those tests are not a good example to follow.

plamut replied:
OK, agreed. Can you please also create a ticket to refactor those tests? Thanks!

ioanpocol commented:
:+1:

plamut commented:
Updated, the test now uses promises all the way down (the likely reason for element matching to fail?). I also rebased onto latest master, there was some unrelated unit test failing for some unknown reason.

plamut commented:
OK, so the thing is we don't navigate to the right page. We are supposed to go to `/authoring` but end up on `/workspace` as can be seen in debug output - https://travis-ci.org/superdesk/superdesk/jobs/74591008#L1266
(FWIW, I've tried both `nav()` and `open()` functions from utils, and they both behave the same)

Are there any redirects going on here or what?

Anyway, will continue looking at this on Monday.

mugurrus commented on client/spec/authoring_spec.js:
```diff
@@ -102,4 +102,38 @@ describe('authoring', function() {
     });
+
+    describe('subnavigation menu', function () {
+        fit('allows to create a new empty package', function () {
```
f needs to be removed

mugurrus replied:
Sorry, saw later that you're still debugging this.

plamut commented:
Resolved in #945, a workaround was made to trigger the package creation action through the `/monitoring` view.
